syntax = "proto3";

message storedata_request
{
    string filename = 1;
}

message getdata_request
{
    string filename = 1;
}

message fragment_status_request
{
    string fragment_name = 1;
}

message fragment_status_response
{
    string fragment_name = 1;
    bool is_present = 2;
    string node_id = 3;
    int32 count = 4;
}

enum request_type
{
    FRAGMENT_STATUS_REQ = 0;
    FRAGMENT_DATA_REQ = 1;
    STORE_FRAGMENT_DATA_REQ = 2;
    RECODE_FRAGMENTS_REQ = 3;
}

// This message is sent in the first frame of the request,
// so the other side knows what format to expect in the second frame
message header
{
    request_type request_type = 1;
}

message recode_fragments_request
{
    string fragment_name = 1;
    int32 symbol_count = 2;
    int32 output_fragment_count = 3;
}

